types/subleveldown/index.d.ts

```diff
@@ -0,0 +1,33 @@
+// Type definitions for subleveldown 4.1
+// Project: https://github.com/level/subleveldown
+// Definitions by: Carson Farmer <https://github.com/carsonfarmer>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { LevelUp } from 'levelup';
+import { ErrorValueCallback, ErrorCallback, AbstractGetOptions } from 'abstract-leveldown';
+
+declare namespace sub {
+    interface Options {
+        separator?: string;
+        open?: ((callback?: ErrorCallback) => any);
+        [key: string]: any;
+    }
+
+    type LevelUpClear<V, O> =
+        ((callback: ErrorValueCallback<V>) => void) &
+        ((options: O, callback: ErrorValueCallback<V>) => void) &
+        ((options?: O) => Promise<V>);
+
+    type InferDBClear<DB> =
+        DB extends { clear: (options: infer O, callback: ErrorValueCallback<infer V>) => void } ?
+        LevelUpClear<V, O> :
+        LevelUpClear<any, AbstractGetOptions>;
+
+    interface SublevelDown<DB, Iterator> extends LevelUp<DB, Iterator> {
+        clear: InferDBClear<DB>;
+    }
+}
+
+declare function sub<DB, Iterator>(db: LevelUp < DB, Iterator >, prefix ?: string, options ?: sub.Options): sub.SublevelDown<DB, Iterator>;
```

Comment:
`SubIterator` is actually a class. I suggest using functions and ambient classes. [Example].

[Example]: https://www.typescriptlang.org/play/#code/MYGwhgzhAECCBGEAuAnMwkEkkFM1IHsUAeAEQCEAaabPMQlAPmgG8BfAWAChuBLAO1woAZuhzQAMjgBuOEAFUADmSo0h9Is3Z9BeUcHEB5RUl4F+MbTy4B6G9AC0T5y4fduAExygwKcT6hoAGUAV3hafCIVagiNJmgcAA9cfg8YBGQ0DFiGaLU6BkYWbmhS6CQAW0UALmgwfgBPAG53LjLoRTCQXmBoYHNMkIwiAAoPeFqpWQVFaDyczWpFP2FeRIB+WsyBAHNqAhMzC2hN6GNTAYBKFq5Oay8fP2hhEP4MI+CwhZIKGPVCsYTSQyORKeb-RYdFZrU7bfh7aAHC4WU7nI4QS61ULhCE-VTfRg3bj9CxIco4ZAARmgAF5oPwcAB3T44gqjdjXaB2FnfVok5Dk5AAJlpPNxIw5TS59mxvOsuCpADpKooiVwFUghcqqk0gA

Reply:
This sounds reasonable. In this case, do you mean the main `SubDown` function/class, or the 'internal' `SubIterator`? In the source code, `SubIterator` is not exported, though it obviously is returned from the `iterator` method. Are you suggesting:
- [ ] Add this `SubIterator` class to the definition as well?
- [ ] Simply implement the main `SubDown` using a function and ambient class?
- [ ] Both of these?

Note that I updated the name of the main interface (`SubDown`) to better match the source code.

Reply:
Either way, your suggestion is a good one for the main class/function/interface. So I will certainly add this.